plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -3467,4 +3471,92 @@ public void testTrinoCacheInvalidatedOnCreateTable()
 
         assertUpdate("DROP TABLE " + tableName);
     }
+
+    @ParameterizedTest
+    @MethodSource("timestampPrecisionOnCreateTableProvider")
+    public void testTimestampPrecisionOnCreateTable(TimestampPrecisionTestSetup setup)
+    {
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "test_coercion_show_create_table",
+                "(ts TIMESTAMP)")) {
+            assertUpdate("INSERT INTO " + testTable.getName() + " VALUES (" + setup.sourceValueLiteral() + ")", 1);
+            assertEquals(getColumnType(testTable.getName(), "ts"), setup.newColumnType());
+            assertQuery("SELECT * FROM " + testTable.getName(), "VALUES " + setup.newValueLiteral());
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("timestampPrecisionOnCreateTableProvider")
```

Comment:
we don't use parametrized tests afaik. please loop over values in the method itself

Reply:
I see that Iceberg is already using parametrized tests https://github.com/trinodb/trino/commit/db9f1b9b3c45194793149b699bcc757e1b79521f

Reply:
Some additional context on not using Parameterized tests https://github.com/trinodb/trino/pull/19532#discussion_r1391669051.

@martint / @dain In this case even if we replace the  parametrized tests with a for loop we still won't be able to properly abstract the test data from the testing code - Or can we replace them with a TypeMapping like tests here ? Which allows us to test multiple input cases and treats them as a single row